templates/serviceaccount.yaml

```
@@ -1,15 +1,17 @@
-{{- if .Values.serviceAccount.enabled -}}
+{{- range $name, $values := .Values.Servers }}
+{{- if $values.serviceAccount -}}
 apiVersion: v1
 kind: ServiceAccount
 metadata:
-  name: {{ include "<CHARTNAME>.fullname" . }}-sa
+  name: {{ include "<CHARTNAME>.fullname" $ }}-{{ $name }}-sa
```

`-sm` suffix is redundant
```suggestion
  name: {{ include "<CHARTNAME>.fullname" $ }}-{{ $name }}
```